src/Microsoft.AspNet.OData.Shared/DefaultContainerBuilder.cs

```diff
@@ -85,9 +85,7 @@ public virtual IServiceProvider BuildContainer()
             * More info at https://github.com/OData/WebApi/pull/1082
             */
 
-            Assembly microsoftExtensionsDependencyInjectionAssembly = services.GetType().GetTypeInfo().Assembly;
-            TypeInfo serviceCollectionContainerBuilderExtensionsType = microsoftExtensionsDependencyInjectionAssembly.GetType(typeof(ServiceCollectionContainerBuilderExtensions).GetTypeInfo().FullName).GetTypeInfo();
-            MethodInfo buildServiceProviderMethod = serviceCollectionContainerBuilderExtensionsType.GetMethod("BuildServiceProvider", new[] { typeof(IServiceCollection) });
+            MethodInfo buildServiceProviderMethod = typeof(ServiceCollectionContainerBuilderExtensions).GetMethod(nameof(ServiceCollectionContainerBuilderExtensions.BuildServiceProvider), new[] { typeof(IServiceCollection) });
```

Comment:
(nit) how often does this get called? I assume only once usually. So it probably doesn't pay to cache it.